File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        boolean[] isVowel = new boolean[128];
        int left = 0; 
        int right = s.length() - 1;
        char[] charArray = s.toCharArray();
        
        for (char c : "aeiouAEIOU".toCharArray()) {
            isVowel[c] = true;
        }

        while (left < right) {

            while (left < right && !isVowel[charArray[left]]) {
                left++;
            } 
            
            while (left < right && !isVowel[charArray[right]]) {
                right--;
            }
            
            if (left < right) {
                char temp = charArray[left];
                charArray[left] = charArray[right];
                charArray[right] = temp;
                left++;
                right--;
            } 
        }
        return String.valueOf(charArray);
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        left, right = 0, len(s) - 1
        vowelList = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        stringList = list(s)

        while left < right:
            if stringList[left] not in vowelList:
                left += 1
                continue
            if stringList[right] not in vowelList:
                right -= 1
                continue
            
            temp = stringList[left]
            stringList[left] = stringList[right]
            stringList[right] = temp
            left += 1
            right -= 1
        
        return ''.join(stringList)
